File: Javascript/58.length-of-last-word.js
/*
 * @lc app=leetcode id=58 lang=javascript
 *
 * [58] Length of Last Word
 */

// @lc code=start
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    s = s.trimEnd();
    const w = s.lastIndexOf(" ");
    let lastWord = s;
    if (w >= 0) {
        lastWord = s.substring(w, s.length - 1);
    }
    return lastWord.length;
};
// @lc code=end

console.log(lengthOfLastWord(" a"))

File: Ruby/58.length-of-last-word.rb
#
# @lc app=leetcode id=58 lang=ruby
#
# [58] Length of Last Word
#

# @lc code=start
# @param {String} s
# @return {Integer}
def length_of_last_word(s)
    word = ""

    s.strip!
    for i in 0...s.size
        if s[i] == " "
            word = ""
        else
            word += s[i]
        end
    end

    return word.size
end
# @lc code=end

